ludwig/data/split.py

```diff
@@ -108,8 +128,21 @@ def split(
         self, df: DataFrame, backend: Backend, random_seed: float = default_random_seed
     ) -> Tuple[DataFrame, DataFrame, DataFrame]:
         if backend.df_engine.partitioned:
-            # TODO dask: find a way to support this method
-            raise ValueError('Split type "stratify" is not supported with a partitioned dataset.')
+            # For a partitioned dataset, we can stratify split each partition individually
+            # to obtain a global stratified split.
+
+            # Default to 0 for train
+            df[TMP_SPLIT_COL] = 0
+
+            # Recursively split each partition
+            partition_splitter = _get_partition_splitter(splitter=self, random_seed=random_seed)
+            df = backend.df_engine.map_partitions(df, partition_splitter, meta=df)
+
+            df_train = df[df[TMP_SPLIT_COL] == 0].drop(columns=TMP_SPLIT_COL)
+            df_val = df[df[TMP_SPLIT_COL] == 1].drop(columns=TMP_SPLIT_COL)
+            df_test = df[df[TMP_SPLIT_COL] == 2].drop(columns=TMP_SPLIT_COL)
+
```

Comment:
As discussed offline, it would be good to repartition the dataframes here and benchmark what the overhead of repartitioning is:

- How expensive is repartitioning
- Is not repartitioning expensive for downstream dask processes? E.g. oversampling the dataset happens after the split [here](https://github.com/ludwig-ai/ludwig/blob/27db097094a86791d13c6740402d02d8439b3d52/ludwig/data/preprocessing.py#L1829). Is oversampling slower/faster/about the same if we repartition?

I don't think there are any other dask preproc steps after we do splitting, so testing oversampling should be sufficient. And after further thought, it should be ok to skip testing if prefetching is affected.

Reply:
Based on benchmarking results for overall preprocessing time with and without repartitioning, it's ok to not repartition.